src/Interpreters/Access/InterpreterGrantQuery.cpp

```diff
@@ -366,6 +366,13 @@ BlockIO InterpreterGrantQuery::execute()
     AccessRightsElements elements_to_grant, elements_to_revoke;
     collectAccessRightsElementsToGrantOrRevoke(query, elements_to_grant, elements_to_revoke);
 
+    if (query.current_grants)
```

Comment:
If `query.current_grants == true` we should also disable `checkGrantOptions` later in this function. Otherwise we will be getting get strange errors if the current grants are changed by a parallel command while `InterpreterGrantQuery::execute()` is running.